x/feerefunder/keeper/keeper_test.go

```diff
@@ -156,18 +163,113 @@ func TestKeeperLockFees(t *testing.T) {
 		TimeoutFee: sdk.NewCoins(sdk.NewCoin("denom1", sdk.NewInt(101))),
 	}
 	channelKeeper.EXPECT().GetChannel(ctx, packet.PortId, packet.ChannelId).Return(channeltypes.Channel{}, true)
-	bankKeeper.EXPECT().SendCoinsFromAccountToModule(ctx, payer, types.ModuleName, validFee.Total()).Return(fmt.Errorf("bank error"))
-	err = k.LockFees(ctx, payer, packet, validFee)
+	bankKeeper.EXPECT().SendCoinsFromAccountToModule(ctx, payerInfo.Sender, types.ModuleName, validFee.Total()).Return(fmt.Errorf("bank error"))
+	err = k.LockFees(ctx, payerInfo, packet, validFee)
 	require.ErrorContains(t, err, "bank error")
 
 	channelKeeper.EXPECT().GetChannel(ctx, packet.PortId, packet.ChannelId).Return(channeltypes.Channel{}, true)
-	bankKeeper.EXPECT().SendCoinsFromAccountToModule(ctx, payer, types.ModuleName, validFee.Total()).Return(nil)
-	err = k.LockFees(ctx, payer, packet, validFee)
+	bankKeeper.EXPECT().SendCoinsFromAccountToModule(ctx, payerInfo.Sender, types.ModuleName, validFee.Total()).Return(nil)
+	err = k.LockFees(ctx, payerInfo, packet, validFee)
+	require.NoError(t, err)
+	require.Equal(t, sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeLockFees,
+			sdk.NewAttribute(types.AttributeKeyPayer, payerInfo.Sender.String()),
+			sdk.NewAttribute(types.AttributeKeyPortID, packet.PortId),
+			sdk.NewAttribute(types.AttributeKeyChannelID, packet.ChannelId),
+			sdk.NewAttribute(types.AttributeKeySequence, strconv.FormatUint(packet.Sequence, 10)),
+		),
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		),
+	}, ctx.EventManager().Events())
+}
+
+func TestKeeperGetPayerInfo(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	bankKeeper := mock_types.NewMockBankKeeper(ctrl)
+	channelKeeper := mock_types.NewMockChannelKeeper(ctrl)
+	feegrantKeeper := mock_types.NewMockFeeGrantKeeper(ctrl)
+	k, ctx := testutil_keeper.FeeKeeper(t, channelKeeper, bankKeeper, feegrantKeeper)
+
+	_, err := k.GetPayerInfo(ctx, "", "")
+	require.ErrorContains(t, err, "failed to parse address")
+
+	p, err := k.GetPayerInfo(ctx, TestAddress, "")
+	require.NoError(t, err)
+	require.Equal(t, p, types.PayerInfo{
+		Sender:   sdk.MustAccAddressFromBech32(TestAddress),
+		FeePayer: nil,
+	})
+
+	p, err = k.GetPayerInfo(ctx, TestAddress, TestAddress)
```

Comment:
To test logic to verify that it's different addresses it would probably be better to send different addresses as args